controllers/subjects.js

```diff
@@ -68,6 +72,16 @@ exports.addSubject = asyncHandler(async (req, res, next) => {
       404
     );
   }
+
+  // Make sure user is category  owner
+  if (category.students.toString() !== req.students.id && req.students.role !== "admin") {
```

Comment:
I really don't understand this aspect of the code. Where did we get the `category` object. We only had the categoryId
Please explain